commitments/mode.go

```diff
@@ -4,51 +4,64 @@ import (
 	"fmt"
 )
 
-type CommitmentMeta struct {
-	Mode CommitmentMode
-	// version is shared for all modes and denotes version of the EigenDA certificate
-	Version EigenDACommitmentType
-}
-
 type CommitmentMode string
 
 const (
-	OptimismKeccak  CommitmentMode = "optimism_keccak256"
-	OptimismGeneric CommitmentMode = "optimism_generic"
-	Standard        CommitmentMode = "standard"
+	OptimismKeccakCommitmentMode  CommitmentMode = "optimism_keccak256"
+	OptimismGenericCommitmentMode CommitmentMode = "optimism_generic"
+	StandardCommitmentMode        CommitmentMode = "standard"
 )
 
 func StringToCommitmentMode(s string) (CommitmentMode, error) {
 	switch s {
-	case string(OptimismKeccak):
-		return OptimismKeccak, nil
-	case string(OptimismGeneric):
-		return OptimismGeneric, nil
-	case string(Standard):
-		return Standard, nil
+	case string(OptimismKeccakCommitmentMode):
+		return OptimismKeccakCommitmentMode, nil
+	case string(OptimismGenericCommitmentMode):
+		return OptimismGenericCommitmentMode, nil
+	case string(StandardCommitmentMode):
+		return StandardCommitmentMode, nil
 	default:
 		return "", fmt.Errorf("unknown commitment mode: %s", s)
 	}
 }
 
 func EncodeCommitment(
-	bytes []byte,
+	versionedCert EigenDAVersionedCert,
 	commitmentMode CommitmentMode,
-	commitmentType EigenDACommitmentType,
 ) ([]byte, error) {
+	serializedCert := versionedCert.SerializedCert
+	certVersion := versionedCert.CertVersion
+
 	switch commitmentMode {
-	case OptimismKeccak:
-		return Keccak256Commitment(bytes).Encode(), nil
+	case OptimismKeccakCommitmentMode:
+		return Keccak256Commitment(serializedCert).Encode(), nil
 
-	case OptimismGeneric:
-		certCommit := NewEigenDACommitment(bytes, commitmentType).Encode()
+	case OptimismGenericCommitmentMode:
+		certCommit := NewEigenDAVersionedCert(serializedCert, certVersion).Encode()
 		svcCommit := EigenDASvcCommitment(certCommit).Encode()
 		altDACommit := NewGenericCommitment(svcCommit).Encode()
+		// Proxy returns an altDACommitment, which doesn't contain the first op version_byte
+		// (from https://specs.optimism.io/experimental/alt-da.html#example-commitments)
+		// This is because its added by op-alt-da when calling TxData() right before submitting the tx:
```

Comment:
its -> it's

Reply:
[d6300c0](https://github.com/Layr-Labs/eigenda-proxy/pull/379/commits/d6300c01fab28306005bfd681d45348a3288ce34)